progenitor-client/src/progenitor_client.rs

```diff
@@ -386,11 +386,23 @@ pub fn encode_path(pc: &str) -> String {
 }
 
 #[doc(hidden)]
-pub trait RequestBuilderExt<E> {
+pub trait RequestBuilderExt<E>
+where
+    Self: Sized,
+{
     fn form_urlencoded<T: Serialize + ?Sized>(
         self,
         body: &T,
     ) -> Result<RequestBuilder, Error<E>>;
+
+    fn form_from_raw<
+        S: AsRef<str>,
+        T: AsRef<[u8]>,
+        I: Sized + IntoIterator<Item = (S, T)>,
```

Comment:
Why `IntoIterator`?

Reply:
Because a form can contain multiple streams. That's the least limiting for an Impl and sufficient for the `R..BuilderExt` trait. But with the comments below this might become obsolete